source/base/mpi.cc

```diff
@@ -812,7 +814,756 @@ namespace Utilities
 #endif
     }
 
+#ifdef DEAL_II_WITH_MPI
+
+#  ifdef DEBUG
+#    define COMPUTE_INDEX_OWNER_USE_OWN_IBARRIER
+#  endif
+
+    namespace ComputeIndexOwner
+    {
+      struct Dictionary
+      {
+        std::vector<unsigned int> array;
```

Comment:
can you give this variable a more meaningful name? what is it supposed to store?